platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ConfigOperation.java

```diff
@@ -34,58 +45,244 @@
+            strategy.write(outputStream, configs);
+        }
+    }
+
+    /**
+     * Transactional handler for creating config files
+     */
+    private static class CreateHandler extends ConfigOperation {
+        private CreateHandler(Path configFile, Dictionary<String, Object> configs,
+                PersistenceStrategy strategy) {
+            super(configFile, configs, strategy);
+        }
+
+        @Override
+        public Result<Void> commit() throws ConfiguratorException {
+            try {
+                createConfig(configFile, strategy, configs);
+                return ResultImpl.pass();
+            } catch (IOException e) {
+                return ResultImpl.fail(new ConfiguratorException("Error creating config: ", e));
+            }
+        }
+
+        /**
+         * If a config was created on top of an already existing file, a successful rollback would
+         * cause the file to be deleted, which was NOT precisely the state the system was in before.
+         * The caller is responsible for invoking the appropriate operation knowing full well the
+         * rollback consequences.
+         */
+        @Override
+        public Result<Void> rollback() throws ConfiguratorException {
+            boolean deleted = configFile.delete();
+            if (!deleted) {
+                LOGGER.debug("Problem deleting properties file {}", configFile);
+                return ResultImpl.rollbackFail(new ConfiguratorException(
+                        "Problem deleting properties file " + configFile));
+            }
+            return ResultImpl.rollback();
+        }
+
+        @Override
+        public String toString() {
+            return "Creating config " + configFile.getAbsolutePath();
+        }
     }
 
     /**
      * Transactional handler for deleting config files
      */
     private static class DeleteHandler extends ConfigOperation {
-        private DeleteHandler(Path configFile) {
-            super(configFile);
+        private Dictionary<String, Object> originalConfigs;
+
+        private DeleteHandler(Path configFile, PersistenceStrategy strategy) {
+            super(configFile, null, strategy);
+            this.originalConfigs = new Hashtable<>();
         }
 
         @Override
         public Result<Void> commit() throws ConfiguratorException {
+            originalConfigs = readConfig(configFile, strategy);
             boolean deleted = configFile.delete();
             if (!deleted) {
                 LOGGER.debug("Problem deleting properties file {}", configFile);
+                return ResultImpl.fail(new ConfiguratorException(
+                        "Problem deleting properties file " + configFile));
             }
             return ResultImpl.pass();
         }
 
         @Override
         public Result<Void> rollback() throws ConfiguratorException {
-            return ResultImpl.rollback();
+            try {
+                createConfig(configFile, strategy, originalConfigs);
```

Comment:
Just like you attempt to restore the original file here on rollback, that's what you should do in the case of an overwrite.